modules/assim.sequential/R/sda.enkf.R

```diff
@@ -179,72 +187,74 @@ sda.enkf <- function(settings, obs.mean, obs.cov, IC = NULL, Q = NULL){
         "hostname    : ", settings$host$name, "\n",
         "rundir      : ", file.path(settings$host$rundir, run.id[[i]]), "\n",
         "outdir      : ", file.path(settings$host$outdir, run.id[[i]]), "\n",
-        file=file.path(settings$rundir, run.id[[i]], "README.txt"), sep='')
-    
+        file = file.path(settings$rundir, run.id[[i]], "README.txt"), 
+        sep='')
   }
   
   ## add the jobs to the list of runs
-  cat(as.character(unlist(run.id)),file=file.path(settings$rundir, "runs.txt"),sep="\n",append=FALSE)
+  cat(as.character(unlist(run.id)), 
+      file = file.path(settings$rundir, "runs.txt"),
+      sep = "\n", 
+      append = FALSE)
   
   ## start model runs
-  start.model.runs(settings,settings$database$bety$write)
-  save.image(file.path(outdir,"sda.initial.runs.Rdata"))
- 
+  start.model.runs(settings, settings$database$bety$write)
+  save.image(file.path(outdir, "sda.initial.runs.Rdata"))
+  
   ###-------------------------------------------------------------------###
   ### tests before data assimilation                                    ###
   ###-------------------------------------------------------------------###  
   
-  # at some point add a lot of error checking
-  # read time from data
-  # if data is missing you still need to have NAs or NULL with date name
-  ## vector to read the correct netcdfs by read.restart
+  # at some point add a lot of error checking 
+  # read time from data if data is missing you still need
+  # to have NAs or NULL with date name vector to read the correct netcdfs by read.restart
   
-  obs.times = names(obs.mean)
-  obs.times.POSIX = ymd_hms(obs.times)
+  obs.times <- names(obs.mean)
+  obs.times.POSIX <- ymd_hms(obs.times)
   
-  for(i in 1:length(obs.times)){
-    if(is.na(obs.times.POSIX[i])){
-      if(is.na(ymd(obs.times[i]))){
-        print('Error: no dates associated with observations')
-      }else{
+  for (i in seq_along(obs.times)) {
```

Comment:
If length(obs.time) is zero your way will loop from 1:0 while Ben's will skip the loop.   The latter is the desired behavior